src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -677,6 +677,58 @@ struct accel_deadlock_status
   }
 };
 
+struct cu_size
+{
+  using result_type = query::cu_size::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& cu_idx)
+  {
+    const auto index = boost::any_cast<uint32_t>(cu_idx);
+    const auto cu_subdev = "CU[" + std::to_string(index) + "]";
+    auto pdev = get_pcidev(device);
+    std::string errmsg;
+    result_type size;
+
+    pdev->sysfs_get(cu_subdev, "size", errmsg, size, static_cast<uint32_t>(0));
+    if (!errmsg.empty())
+      throw xrt_core::query::sysfs_error(errmsg);
+
+    return size;
+  }
+};
+
+struct cu_read_range
+{
+  using result_type = query::cu_read_range::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& cu_idx)
+  {
+    using tokenizer = boost::tokenizer< boost::char_separator<char> >;
+    const auto index = boost::any_cast<uint32_t>(cu_idx);
+    const auto cu_subdev = "CU[" + std::to_string(index) + "]";
+    auto pdev = get_pcidev(device);
+    std::string errmsg;
+    std::string range_str;
+    result_type range = {0, 0};
+
+    // The read_range is printing in formatted string
+    // Format: "%d %d"
+    pdev->sysfs_get(cu_subdev, "read_range", errmsg, range_str);
+    if (!errmsg.empty())
+      throw xrt_core::query::sysfs_error(errmsg);
+
+    tokenizer tokens(range_str);
+    const int radix = 16;
+    tokenizer::iterator tok_it = tokens.begin();
+    range.start = std::stoul(std::string(*tok_it++), nullptr, radix);
+    range.end = std::stoul(std::string(*tok_it++), nullptr, radix);
+
+    return range;
+  }
+};
```

Comment:
This doesn't feel right.  The parsing should be at coreutil level in query_request.
This file should not need changes beyond adding (emplacing) the two new sysfs query requests.
You may want to check 
core/common/query.h:85 (get(..., modifier, const std::string& modifier);
core/common/query_requests.h:2410 (for a request that uses subdev modifier)
The idea would then be to simply call `get(...,modifier::subdev, subdev)` where subdev is for example `CU[0]` or what have you.  The query_request structure would have a parse function in cu_read_range case.

I didn't exactly work through this, but what you have here doesn't scale very well beyond Alveo Linux.

Reply:
Let me double check. Thanks for the pointer!